docs/content/docs/reference/react/hooks.mdx

```diff
@@ -0,0 +1,393 @@
+---
+title: 'Hooks'
+---
+
+## Authentication Hooks
+
+### Email + password registration
+
+```js
+const { signUp, isLoading, isSuccess, needsVerification, isError, error } =
+  useEmailPasswordSignUp(email?: string, password?: string, options?: SignUpOptions )
+```
+
+- `signUp: (email?: string, password?: string) => void` triggers the user registration. The email/password arguments will take precedence over the possible state values used when creating the hook.
+- `isLoading: boolean` returns `true` when the action is executing, `false` when the action result is available.
+- `needsVerification: boolean` returns `true` if the registration has been accepted, but a verificaiton email has been sent and is awaiting.
+- `isSuccess: boolean` returns `true` if the registration suceeded. Returns `false` if the new email needs to be verified first, or if an error occurred.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+- `SignUpOption`:
+  - `locale?: string` locale of the user, in two digits, for instance `en`.
+  - `allowedRoles?: string[]`: allowed roles of the user. Must be a subset of the default allowed roles defined in Hasura Auth.
+  - `defaultRole?: string`: default role of the user. Must be part of the default allowed roles defined in Hasura Auth.
+  - `displayName?: string`
+  - `metadata?: Record<string, unknown>`: Custom additional user information stored in the `metadata` column.
+  - `redirectTo?: string`: redirection path in the client application that will be used in the link in the verification email. For instance, if you want to redirect to `https://myapp.com/success`, the `redirectTo` value is `'/success'`.
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useEmailPasswordSignUp } from '@nhost/react'
+
+const Component = () => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const { signUp, isLoading, isSuccess, needsVerification, isError, error } =
+    useEmailPasswordSignUp(email, password)
+  return (
+    <div>
+      <input value={email} onChange={(event) => setEmail(event.target.value)} placeholder="Email" />
+      <input
+        value={password}
+        onChange={(event) => setPassword(event.target.value)}
+        placeholder="Password"
+      />
+      <button onClick={signUp}>Register</button>
+      {isSuccess && <div>Your account have beed created! You are now authenticated</div>}
+      {needsVerification && (
+        <div>
+          Please check your mailbox and follow the verification link to confirm your registration
+        </div>
+      )}
+    </div>
+  )
+}
+```
+
+### Email + password authentication
```

Comment:
Suggestion: 

### Email + password authentication

to

### Email and Password Sign-In